src/main/java/com/google/sps/Trip.java

```diff
@@ -0,0 +1,129 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import java.util.ArrayList;
+
+/**
+ * Trip is the class for storing a single trip (could be multiple days).
+ */
+public class Trip {
+  private String tripName;
+  private String startDate;
+  private String endDate;
+  private int numDays;
+  private ArrayList<TripDay> tripDays;
+
+  /**
+   * Creates a new Trip.
+   *
+   * @param tripName The human-readable name for the trip. Must be non-null.
+   * @param startDate The start date for the trip. Must be non-null.
+   * @param endDate The end date for the trip. Must be non-null.
+   * @param numDays The number of days for the trip. 
+   * @param tripDays The list of tripDays. Must be non-null.
+   */
+  public Trip(String tripName, String startDate, String endDate, int numDays, ArrayList<TripDay> tripDays) {
+    if (tripName == null) {
+      throw new IllegalArgumentException("tripName cannot be null");
+    }
+
+    if (startDate == null) {
+      throw new IllegalArgumentException("startDate cannot be null");
+    }
+
+    if (endDate == null) {
+      throw new IllegalArgumentException("endDate cannot be null");
+    }
+
+    if (tripDays == null) {
+      throw new IllegalArgumentException("tripDays cannot be null. Use empty array instead.");
+    }
+
+    this.tripName = tripName;
+    this.startDate = startDate;
+    this.endDate = endDate;
+    this.numDays = numDays;
```

Comment:
I suggest adding a check that number of days cannot be less than 1 and that there is also a max, say one month (30 days)

Reply:
Building on this, this may be excess / unnecessary, but I'm wondering if you could add a check to determine whether the `numDays` value is consistent with the `startDate` and `endDate` variables. This may be difficult given that those are Strings though, so I'll leave this up to whatever you decide.

Reply:
Added a check to ensure the number of days is between 1 and 31 (inclusive), and am now computing the number of days using the SimpleDateFormat library so that it is consistent with the startDate and endDate.